app/screens/permalink/permalink.js

```diff
@@ -0,0 +1,455 @@
+        intl: intlShape.isRequired,
+    };
+
+    constructor(props) {
+        super(props);
+
+        const {postIds, channelName} = props;
+        let loading = false;
+
+        if (postIds && postIds.length >= 10) {
+            loading = true;
+        }
+
+        this.state = {
+            title: channelName,
+            loading,
+            error: '',
+            retry: false,
+        };
+    }
+
+    componentWillMount() {
+        if (!this.state.loading) {
+            this.loadPosts(this.props);
+        }
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (this.props.channelName !== nextProps.channelName) {
+            this.setState({title: nextProps.channelName});
+        }
+
+        if (this.props.focusedPostId !== nextProps.focusedPostId) {
+            this.setState({loading: false});
+            if (nextProps.postIds && nextProps.postIds.length < 10) {
+                this.loadPosts(nextProps);
+            } else {
+                this.setState({loading: true});
+            }
+        }
+    }
+
+    goToThread = preventDoubleTap((post) => {
+        const {actions, navigator, theme} = this.props;
+        const channelId = post.channel_id;
+        const rootId = (post.root_id || post.id);
+
+        actions.loadThreadIfNecessary(rootId, channelId);
+        actions.selectPost(rootId);
+
+        const options = {
+            screen: 'Thread',
+            animated: true,
+            backButtonTitle: '',
+            navigatorStyle: {
+                navBarTextColor: theme.sidebarHeaderTextColor,
+                navBarBackgroundColor: theme.sidebarHeaderBg,
+                navBarButtonColor: theme.sidebarHeaderTextColor,
+                screenBackgroundColor: theme.centerChannelBg,
+            },
+            passProps: {
+                channelId,
+                rootId,
+            },
+        };
+
+        navigator.push(options);
+    });
+
+    handleClose = () => {
+        const {actions, navigator, onClose} = this.props;
+        if (this.refs.view) {
+            this.refs.view.zoomOut().then(() => {
+                actions.selectPost('');
+                navigator.dismissModal({animationType: 'none'});
+
+                if (onClose) {
+                    onClose();
+                }
+            });
+        }
+    };
+
+    jumpToChannel = (channelId, channelDisplayName) => {
+        if (channelId) {
+            const {actions, channelTeamId, currentTeamId, navigator, onClose, theme} = this.props;
+            const currentChannelId = this.props.channelId;
+            const {
+                handleSelectChannel,
+                handleTeamChange,
+                markChannelAsRead,
+                setChannelLoading,
+                setChannelDisplayName,
+                markChannelAsViewed,
+            } = actions;
+
+            actions.selectPost('');
+
+            if (onClose) {
+                onClose();
+            }
+
+            navigator.resetTo({
```

Comment:
In regards to https://github.com/mattermost/mattermost-mobile/pull/1449#discussion_r169676174, I think we should change back to popToRoot here if we can. There's no sense in redefining all this since it could easily be forgotten later on if we change how the root is defined.

Reply:
It does not work with popToRoot try it out plz

Reply:
and you may think that `popToRoot` is enough as I did as well, but when testing you see how it is not